scripts/py_matter_idl/matter_idl/matter_idl_parser.py

```diff
@@ -524,9 +535,84 @@ def c_comment(self, token: Token):
             self.doc_comments.append(PrefixCppDocComment(token))
 
 
+def _referenced_type_names(cluster: Cluster) -> Set[str]:
+    """
+    Return the names of all data types referenced by the given cluster.
+    """
+    types = set()
+    for s in cluster.structs:
+        for f in s.fields:
+            types.add(f.data_type.name)
+
+    for e in cluster.events:
+        for f in e.fields:
+            types.add(f.data_type.name)
+
+    for a in cluster.attributes:
+        types.add(a.definition.data_type.name)
+
+    return types
+
+
+class GlobalMapping:
+    """
+    Maintains global type mapping from an IDL
+    """
+
+    def __init__(self, idl: Idl):
+        self.bitmap_map = {b.name: b for b in idl.global_bitmaps}
+        self.enum_map = {e.name: e for e in idl.global_enums}
+        self.struct_map = {s.name: s for s in idl.global_structs}
+
+        self.global_types = set(self.bitmap_map.keys()).union(set(self.enum_map.keys())).union(set(self.struct_map.keys()))
```

Comment:
Are these names guaranteed to be unique?

Reply:
Not by spec :( . 

Conceptually they must be because otherwise if I have `enum Foo` and `bitmap Foo` both defined, then when I have somewhere `cluster Bar { attribute Foo foo = 1 }` it is unclear how to determine what it is. Even zap I thin has `type=....` and we do not actually say what namespace to use.

I will add a check.

Reply:
By spec, type names really do have to be unique.  Otherwise referencing types by name does not work.